drake/solvers/SNOPTWrapper.cpp

```diff
@@ -51,7 +51,7 @@ static int snopt_userfun(snopt::integer *Status, snopt::integer *n, snopt::doubl
   memset(G,0,sizeof(*n)*sizeof(snopt::doublereal));
 
   // evaluate objective
-  auto tx = initTaylorVecXd(xvec);
+  auto tx = initializeAutoDiff(xvec);
```

Comment:
We need to decide and agree on what to call all of these things.  My feeling is that having matrix types defined for AutoDiff helps a lot - i had been calling them TaylorVec, TaylorMat, ... I'm ok changing them, but I'm not ok with you writing everything one way and me another.  I think the function signatures should match the types.